src/Traits/AllowedLinkClassesTrait.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace SilverStripe\LinkField\Traits;
+
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Core\Config\Config;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\LinkField\Models\Link;
+use SilverStripe\LinkField\Services\LinkTypeService;
+
+trait AllowedLinkClassesTrait
+{
+    protected $allowed_types = [];
+
+    /**
+     * Set allowed types for LinkField
+     */
+    public function setAllowedTypes(array $types): static
+    {
+        if (!is_array($types) || empty($types) || !$this->validateTypes($types)) {
```

Comment:
We don't need to explicitly test that this is an array. If you call the method with something other than an array, you'll get an exception right away.
```suggestion
        if (empty($types) || !$this->validateTypes($types)) {
```

Reply:
Updated